File: examples/test-terry-davis.java
/**
 * Terry Davis style Java - unconventional, playful, surprising
 * "God said this should work"
 * Compile with: javac test-terry-davis.java
 * Run with: java TerryDavisDemo
 */

import java.util.*;

// God said this should work
public class TerryDavisDemo {
    
    // Cosmic variable names
    static Object GodPointer = null;
    static int[] DivineArray = {1, 2, 3, 4, 5};
    static int HolyCVariable = 42;
    static final String TempleOSKernel = "TempleOS 5.017";
    static List<Integer> SatanBuffer = new ArrayList<>();
    static int HeavenlyLoop = 0;
    
    // WARNING: Satan tries to corrupt this buffer
    public static int[] SacredFunction(int n) {
        // God said this should work
        GodPointer = new int[n];
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            result[i] = i * i;
        }
        return result;
    }
    
    // Goto-style control flow (Java doesn't have goto, but we can simulate)
    public static void ProcessDataWithGoto() {
        System.out.println("The chosen path");
        
        // Simulate goto with labeled blocks
        step2: {
            if (Math.random() > 0.5) {
                System.out.println("God's path not taken");
                break step2;
            }
            System.out.println("Blessed completion");
        }
    }
    
    // Recursive function - infinite temple recursion
    public static void RecursiveTemple(int depth, int maxDepth) {
        if (depth < maxDepth) {
            System.out.println("Temple level " + depth);
            RecursiveTemple(depth + 1, maxDepth);
        } else {
            System.out.println("We have reached the inner sanctum");
        }
    }
    
    // Obfuscated one-liner
    public static void ObfuscatedOneLiner() {
        // Print 0-9 with stream
        System.out.println(IntStream.range(0, 10).mapToObj(String::valueOf).reduce((a, b) -> a + " " + b).orElse(""));
    }
    
    // Macro-like functions (using methods)
    public static boolean GOD_MODE() {
        return true;
    }
    
    public static boolean SATAN_MODE() {
        return false;
    }
    
    public static int TEMPLE(int x) {
        return x * x + 0x666;
    }
    
    // Direct memory manipulation (using arrays)
    public static void DirectMemoryDemo() {
        int[] memory = new int[10];
        
        // Write to "memory"
        for (int i = 0; i < memory.length; i++) {
            memory[i] = i * 100;
        }
        
        // Read from "memory"
        System.out.print("Memory contents: ");
        for (int val : memory) {
            System.out.print(val + " ");
        }
        System.out.println();
    }
    
    // Prototype manipulation (using interfaces and dynamic proxies)
    interface DivineBehavior {
        int sacred(int x);
        String temple(String s);
    }
    
    public static void PrototypeDemo() {
        // Create dynamic proxy to add behavior to any object
        DivineBehavior proxy = (DivineBehavior) java.lang.reflect.Proxy.newProxyInstance(
            DivineBehavior.class.getClassLoader(),
            new Class[]{DivineBehavior.class},
            (obj, method, args) -> {
                if (method.getName().equals("sacred")) {
                    return (int) args[0] * (int) args[0];
                } else if (method.getName().equals("temple")) {
                    return "Temple: " + args[0];
                }
                return null;
            }
        );
        
        System.out.println("Sacred(5) = " + proxy.sacred(5));
        System.out.println("Temple(\"test\") = " + proxy.temple("test"));
    }
    
    // Main execution
    public static void main(String[] args) {
        System.out.println("Terry Davis style Java demonstration");
        System.out.println("===================================");
        System.out.println();
        
        System.out.println("Operating System: " + TempleOSKernel);
        System.out.println("HolyC Variable: " + HolyCVariable);
        
        // God mode activated
        if (GOD_MODE()) {
            System.out.println("God mode activated");
        }
        
        // This never runs
        if (SATAN_MODE()) {
            System.out.println("This never runs");
        }
        
        System.out.println("Temple value: " + TEMPLE(5));
        
        // Sacred function
        int[] result = SacredFunction(5);
        System.out.print("Sacred array: ");
        for (int val : result) {
            System.out.print(val + " ");
        }
        System.out.println();
        
        // Process data with goto simulation
        ProcessDataWithGoto();
        
        // Recursive temple
        RecursiveTemple(0, 5);
        
        // Divine array
        DivineArray = Arrays.copyOf(DivineArray, DivineArray.length + 1);
        DivineArray[DivineArray.length - 1] = 6;
        System.out.print("Divine array: ");
        for (int val : DivineArray) {
            System.out.print(val + " ");
        }
        System.out.println();
        
        // Direct memory demo
        DirectMemoryDemo();
        
        // Prototype demo
        PrototypeDemo();
        
        // Obfuscated one-liner
        ObfuscatedOneLiner();
    }
}

File: examples/test-terry-davis.c
/**
 * Terry Davis style C - direct hardware access, goto spaghetti, recursive main
 * "God said this should work"
 * Compile with: gcc -o terry_davis test-terry-davis.c
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// God said this should work
#define GOD_MODE if(1)
#define SATAN_MODE while(0)
#define TEMPLE(x) (x*x + 0x666)

// Cosmic variable names
int* GodPointer = NULL;
int DivineArray[5] = {1, 2, 3, 4, 5};
int HolyCVariable = 42;
const char* TempleOSKernel = "TempleOS 5.017";
int SatanBuffer[100];
int HeavenlyLoop = 0;
int PropheticFunction(int x) { return x * x + 0x666; }

// WARNING: Satan tries to corrupt this buffer
void SacredFunction(int n) {
    // God said this should work
    GodPointer = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        GodPointer[i] = i * i;
    }
}

// Goto spaghetti - divine redirection
void ProcessData() {
    goto step2;
step1:
    printf("God's path not taken\n");
    goto end;
step2:
    printf("The chosen path\n");
    if (rand() % 2) goto step1;
    printf("Blessed completion\n");
end:
    return;
}

// Recursive main - infinite temple recursion
int main(int argc, char** argv) {
    static int depth = 0;
    if (depth++ < 10) {
        printf("Temple level %d\n", depth);
        main(argc, argv);  // Recursive call
    }
    return 0;
}

// Direct video buffer manipulation - God sees all pixels
void DrawDivinePattern() {
    // In real TempleOS, this would write to video memory
    // For demonstration, we'll just print ASCII art
    printf("Drawing divine pattern:\n");
    for (int y = 0; y < 5; y++) {
        for (int x = 0; x < 20; x++) {
            if ((x + y) % 2 == 0) {
                printf("█");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

// Inline assembly (x86_64) - direct hardware access
void InlineAssemblyDemo() {
    printf("Inline assembly demo:\n");
    
    // This is actual x86_64 assembly that prints a character
    // Note: This is platform-specific and may not work on all systems
    #if defined(__x86_64__) || defined(_M_X64)
        char c = 'A';
        asm volatile (
            "mov $1, %%rax\n"    // sys_write
            "mov $1, %%rdi\n"    // stdout
            "mov %0, %%rsi\n"    // pointer to character
            "mov $1, %%rdx\n"    // length
            "syscall\n"
            : : "r" (&c) : "rax", "rdi", "rsi", "rdx"
        );
        printf("\n");
    #else
        printf("Inline assembly only available on x86_64\n");
    #endif
}

// Macro magic
#define SACRED_LOOP for(int i = 0; i < 5; i++)
#define DIVINE_PRINT(fmt, ...) printf("[DIVINE] " fmt, ##__VA_ARGS__)

void MacroMagicDemo() {
    DIVINE_PRINT("Macro magic demonstration\n");
    
    SACRED_LOOP {
        printf("Sacred iteration %d\n", i);
    }
    
    // Macros that do unexpected things
    #define SWAP(a, b) do { typeof(a) temp = a; a = b; b = temp; } while(0)
    
    int x = 42, y = 666;
    DIVINE_PRINT("Before swap: x = %d, y = %d\n", x, y);
    SWAP(x, y);
    DIVINE_PRINT("After swap: x = %d, y = %d\n", x, y);
}

// String literal printing (HolyC style)
void HolyCStringDemo() {
    "HolyC string literal printing\n";
    "This string is automatically printed\n";
    "Value: %d\n", 42;
}

// Function pointer table (like a jump table)
void FunctionPointerTableDemo() {
    typedef void (*DivineFunction)(void);
    
    DivineFunction functions[] = {
        DrawDivinePattern,
        InlineAssemblyDemo,
        MacroMagicDemo,
        HolyCStringDemo
    };
    
    int numFunctions = sizeof(functions) / sizeof(functions[0]);
    
    printf("Function pointer table:\n");
    for (int i = 0; i < numFunctions; i++) {
        printf("Calling function %d:\n", i);
        functions[i]();
        printf("\n");
    }
}

// Bit manipulation (direct hardware style)
void BitManipulationDemo() {
    printf("Bit manipulation demo:\n");
    
    unsigned int divineValue = 0xDEADBEEF;
    printf("Divine value: 0x%X\n", divineValue);
    
    // Set bit 3
    divineValue |= (1 << 3);
    printf("After setting bit 3: 0x%X\n", divineValue);
    
    // Clear bit 3
    divineValue &= ~(1 << 3);
    printf("After clearing bit 3: 0x%X\n", divineValue);
    
    // Toggle bit 3
    divineValue ^= (1 << 3);
    printf("After toggling bit 3: 0x%X\n", divineValue);
}

// Main execution (non-recursive version for demonstration)
int main_demo(int argc, char** argv) {
    printf("Terry Davis style C demonstration\n");
    printf("================================\n\n");
    
    printf("Operating System: %s\n", TempleOSKernel);
    printf("HolyC Variable: %d\n", HolyCVariable);
    
    // God mode activated
    GOD_MODE {
        printf("God mode activated\n");
    }
    
    // This never runs
    SATAN_MODE {
        printf("This never runs\n");
    }
    
    printf("Temple value: %d\n", TEMPLE(5));
    
    // Sacred function
    SacredFunction(5);
    printf("Sacred array: ");
    for (int i = 0; i < 5; i++) {
        printf("%d ", GodPointer[i]);
    }
    printf("\n");
    
    // Process data with goto spaghetti
    ProcessData();
    
    // Draw divine pattern
    DrawDivinePattern();
    
    // Inline assembly demo
    InlineAssemblyDemo();
    
    // Macro magic demo
    MacroMagicDemo();
    
    // HolyC string demo
    HolyCStringDemo();
    
    // Function pointer table demo
    FunctionPointerTableDemo();
    
    // Bit manipulation demo
    BitManipulationDemo();
    
    // Clean up
    free(GodPointer);
    
    printf("\nDivine array: ");
    for (int i = 0; i < 5; i++) {
        printf("%d ", DivineArray[i]);
    }
    printf("\n");
    
    return 0;
}